samples/detect.v2beta1.js

```diff
@@ -0,0 +1,848 @@
+      console.log(`There are ${answers.length} anwser(s);`);
+      answers.forEach(a => {
+        console.log(`   answer: ${a.answer}`);
+        console.log(`   confidence: ${a.matchConfidence}`);
+        console.log(`   match confidence level: ${a.matchConfidenceLevel}`);
+      });
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END dialogflow_detect_intent_knowledge]
+}
+
+function detectIntentwithModelSelection(
+  projectId,
+  sessionId,
+  audioFilePath,
+  languageCode,
+  model
+) {
+  // [START dialogflow_detect_intent_with_model_selection]
+  const fs = require('fs');
+
+  // Imports the Dialogflow client library
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow client.
+  const sessionClient = new dialogflow.SessionsClient();
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = `your GCLOUD project name, if not specified as env variable e.g. my-Gcloud-Project`;
+  // const sessionId = `user specific ID of session, e.g. 12345`;
+  // const audioFilePath = `path to local audio file, e.g. ./resources/book_a_room.wav`;
+  // const languageCode = 'BCP-47 language code, e.g. en-US';
+  // const model = `speech mode selected for given request, e.g. video, phone_call, command_and_search, default`;
+
+  // Define session path
+  const sessionPath = sessionClient.sessionPath(projectId, sessionId);
+  // Read the content of the audio file and send it as part of the request.
+  const readFile = common.util.promisify(fs.readFile, {singular: true});
+  readFile(audioFilePath)
+    .then(inputAudio => {
+      // The audio query request
+      const request = {
+        session: sessionPath,
+        queryInput: {
+          audioConfig: {
+            audioEncoding: `LINEAR16`,
+            sampleRateHertz: 16000,
+            languageCode: languageCode,
+            model: model,
+          },
+        },
+        inputAudio: inputAudio,
+      };
+      // Recognizes the speech in the audio and detects its intent.
+      return sessionClient.detectIntent(request);
+    })
+    .then(responses => {
+      logQueryResult(sessionClient, responses[0].queryResult);
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END dialogflow_detect_intent_with_model_selection]
+}
+
+function logQueryResult(sessionClient, result) {
+  // Imports the Dialogflow library
+  const dialogflow = require('dialogflow');
+
+  // Instantiates a context client
+  const contextClient = new dialogflow.ContextsClient();
+
+  console.log(`  Query: ${result.queryText}`);
+  console.log(`  Response: ${result.fulfillmentText}`);
+  if (result.intent) {
+    console.log(`  Intent: ${result.intent.displayName}`);
+  } else {
+    console.log(`  No intent matched.`);
+  }
+  const parameters = JSON.stringify(
+    structjson.structProtoToJson(result.parameters)
```

Comment:
this is also at detect.js ln 251 - I believe there was a note regarding an error in GRPC that requires structProtoToJson to avoid null values, lemme see if I can find that-

Reply:
What sample is this? There is no region tag

Reply:
After discussing, it sounds like this sample is not used.

Please delete!